sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -1029,3 +1030,110 @@ def test_linalg_warning_with_newton_solver(global_random_seed):
     assert penalized_collinear_newton_deviance == pytest.approx(
         original_newton_deviance, rel=rtol
     )
+
+
+@pytest.mark.parametrize("verbose", [0, 1, 2])
+def test_newton_solver_verbosity(capsys, verbose):
+    """Test the std output of verbose newton solvers."""
+    y = np.array([1, 2], dtype=float)
+    X = np.array([[1.0, 0], [0, 1]], dtype=float)
+    linear_loss = LinearModelLoss(base_loss=HalfPoissonLoss(), fit_intercept=False)
+    sol = _NewtonCholeskySolver(
+        coef=linear_loss.init_zero_coef(X),
+        linear_loss=linear_loss,
+        l2_reg_strength=0,
+        verbose=verbose,
+    )
+    sol.solve(X, y, None)  # returns array([0., 0.69314758])
```

Comment:
Should we assert what is written in the comment?

```suggestion
    optim_coef = sol.solve(X, y, None)
    assert_allclose(optim_coef, array([0., 0.69314758]))
```

Reply:
I don't think so. That would be a self fulfilling prophecy:smirk:
It's meant as info to better understand the choices of Newton steps further down.